CHANGELOG.md

```diff
@@ -4,6 +4,25 @@
 <!--- Distributed under the Boost Software License, Version 1.0. (See accompanying -->
 <!--- file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt) -->
 
+## 0.21.0 (2023-12-06)
+
+### New features
+
+- A new sender adaptor `require_started` allows to detect unstarted senders. ([#869](https://github.com/pika-org/pika/pull/869))
+- The conversion from `any_sender` to `unique_any_sender` has been optimized, reusing the same
+  storage. ([#844](https://github.com/pika-org/pika/pull/844))
+- Limit the number of streams created in `cuda_pool`. ([#864](https://github.com/pika-org/pika/pull/864))
```

Comment:
Same comment here about style: maybe something in the direction of "The number of streams created by the `cuda_pool` is now proportional to the number of threads used by the runtime instead of `hardware_concurrency`"?